chain/chain/src/resharding/manager.rs

```diff
@@ -23,15 +27,27 @@ pub struct ReshardingManager {
     /// A handle that allows the main process to interrupt resharding if needed.
     /// This typically happens when the main process is interrupted.
     pub resharding_handle: ReshardingHandle,
+    /// Takes care of performing resharding on the flat storage.
+    pub flat_storage_resharder: Option<FlatStorageResharder>,
 }
 
 impl ReshardingManager {
     pub fn new(
         store: Store,
         epoch_manager: Arc<dyn EpochManagerAdapter>,
+        runtime_adapter: Arc<dyn RuntimeAdapter>,
         resharding_config: MutableConfigValue<ReshardingConfig>,
+        resharding_sender: Option<ReshardingSender>,
     ) -> Self {
-        Self { store, epoch_manager, resharding_config, resharding_handle: ReshardingHandle::new() }
+        let resharding_handle = ReshardingHandle::new();
+        let flat_storage_resharder = resharding_sender.map(|sender| {
+            FlatStorageResharder::new(
+                runtime_adapter,
+                sender.into_sender(),
+                FlatStorageResharderController::from_resharding_handle(resharding_handle.clone()),
```

Comment:
I'm still unsure how does the controller work and what do we do with the sender and receiver here? I'll have to go back and do a quick read up on how crossbeam_channels work but we should ideally not be introducing a new framework here given we have actors and message senders? Thoughts?

Reply:
I planned to remove the channels in another PR because I thought they could come in handy but they did not. I can push the change here